tests/network/test_services.py

```diff
@@ -100,96 +101,99 @@ def test_get_unspent_test_failure(self):
 
     def test_get_transactions_test_used(self):
-        assert len(CashExplorerBitcoinDotComAPI.get_transactions_testnet(TEST_ADDRESS_USED2)) >= 444
+        assert len(BitcoinDotComAPI.get_transactions_testnet(TEST_ADDRESS_USED2)) >= 444
 
     def test_get_transactions_test_unused(self):
-        assert len(CashExplorerBitcoinDotComAPI.get_transactions_testnet(TEST_ADDRESS_UNUSED)) == 0
+        assert len(BitcoinDotComAPI.get_transactions_testnet(TEST_ADDRESS_UNUSED)) == 0
 
     def test_get_unspent_return_type(self):
-        assert iter(CashExplorerBitcoinDotComAPI.get_unspent(MAIN_ADDRESS_USED1))
+        assert iter(BitcoinDotComAPI.get_unspent(MAIN_ADDRESS_USED1))
 
     def test_get_unspent_main_used(self):
-        assert len(CashExplorerBitcoinDotComAPI.get_unspent(MAIN_ADDRESS_USED2)) >= 1
+        assert len(BitcoinDotComAPI.get_unspent(MAIN_ADDRESS_USED2)) >= 1
 
     def test_get_unspent_main_unused(self):
-        assert len(CashExplorerBitcoinDotComAPI.get_unspent(MAIN_ADDRESS_UNUSED)) == 0
+        assert len(BitcoinDotComAPI.get_unspent(MAIN_ADDRESS_UNUSED)) == 0
 
     def test_get_unspent_test_used(self):
-        assert len(CashExplorerBitcoinDotComAPI.get_unspent_testnet(TEST_ADDRESS_USED2)) >= 194
+        assert len(BitcoinDotComAPI.get_unspent_testnet(TEST_ADDRESS_USED2)) >= 194
 
     def test_get_unspent_test_unused(self):
-        assert len(CashExplorerBitcoinDotComAPI.get_unspent_testnet(TEST_ADDRESS_UNUSED)) == 0
+        assert len(BitcoinDotComAPI.get_unspent_testnet(TEST_ADDRESS_UNUSED)) == 0
 
 
 @decorate_methods(catch_errors_raise_warnings, NetworkAPI.IGNORED_ERRORS)
-class TestBlockdozerAPI:
+class TestBitcoreAPI:
     def test_get_balance_return_type(self):
-        assert isinstance(BlockdozerAPI.get_balance(MAIN_ADDRESS_USED1), int)
+        assert isinstance(BitcoreAPI.get_balance(MAIN_ADDRESS_USED1), int)
 
     def test_get_balance_main_used(self):
-        assert BlockdozerAPI.get_balance(MAIN_ADDRESS_USED1) > 0
+        assert BitcoreAPI.get_balance(MAIN_ADDRESS_USED1) > 0
 
     def test_get_balance_main_unused(self):
-        assert BlockdozerAPI.get_balance(MAIN_ADDRESS_UNUSED) == 0
+        assert BitcoreAPI.get_balance(MAIN_ADDRESS_UNUSED) == 0
 
     def test_get_balance_test_used(self):
-        assert BlockdozerAPI.get_balance_testnet(TEST_ADDRESS_USED2) > 0
+        assert BitcoreAPI.get_balance_testnet(TEST_ADDRESS_USED2) > 0
 
     def test_get_balance_test_unused(self):
-        assert BlockdozerAPI.get_balance_testnet(TEST_ADDRESS_UNUSED) == 0
+        assert BitcoreAPI.get_balance_testnet(TEST_ADDRESS_UNUSED) == 0
 
     def test_get_transactions_return_type(self):
-        assert iter(BlockdozerAPI.get_transactions(MAIN_ADDRESS_USED1))
+        assert iter(BitcoreAPI.get_transactions(MAIN_ADDRESS_USED1))
 
-    def test_get_transactions_main_used(self):
-        assert len(BlockdozerAPI.get_transactions(MAIN_ADDRESS_USED1)) >= 218
+    # FIXME: Bitcore.io only returns 10 elements
```

Comment:
Wait, bitcore.io only returns 10 unspents? Are they the most recent? Does it support pagination?

Does it have a regular balance endpoint? In this case balances will be way off on wallets with more than 10 UTXOs.

This is pretty touchy behavior good thing we got the bitcoin.com API in front.

Reply:
As mentioned in DM, I can't find a pagination system in Bitcore.
If seems to be the 10 most recent elements.

Here is an example:
https://trest.bitcoin.com/v2/address/details/bchtest:qprralpnpx6zrx3w2aet97u0c6rcfrlp8v6jenepj5
https://api.bitcore.io/api/BCH/testnet/address/qprralpnpx6zrx3w2aet97u0c6rcfrlp8v6jenepj5

Reply:
If that's OK with you I will mark this as resolved for the time being as Bitcore is not functional at this point (balance and UTXOs are broken even for wallets with less than 10 UTXOs).
If they do not fix these issues before I think there is no need to look for a pagination system in their code.